x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountService.java

```diff
@@ -101,6 +102,13 @@ public void authenticateToken(ServiceAccountToken serviceAccountToken, String no
                 return;
             }
 
+            if (serviceAccountToken.getSecret().length() < MIN_TOKEN_SECRET_LENGTH) {
+                logger.debug("the length of a service account token's secret value must be at least [{}], but is [{}]",
+                    MIN_TOKEN_SECRET_LENGTH, serviceAccountToken.getSecret().length());
```

Comment:
I think this log message should
1. include the qualifiedName of the token.
2. be clear that authentication has failed for this reason
3. be clear that we're validating the input, not the stored value (we don't know whether the input matches the stored value).
```suggestion
                logger.debug("failing authentication for service account token [{}],"
                        + " the provided credential has length [{}]"
                        + " but a token's secret value must be at least [{}] characters",
                    serviceAccountToken.getQualifiedName(),
                    serviceAccountToken.getSecret().length(),
                    MIN_TOKEN_SECRET_LENGTH);
```